sway/desktop/xdg_shell.c

```diff
@@ -87,32 +63,53 @@ static void popup_unconstrain(struct sway_xdg_popup *popup) {
 	wlr_xdg_popup_unconstrain_from_box(wlr_popup, &output_toplevel_sx_box);
 }
 
-static struct sway_xdg_popup *popup_create(
-		struct wlr_xdg_popup *wlr_popup, struct sway_view *view) {
+static struct sway_xdg_popup *popup_create(struct wlr_xdg_popup *wlr_popup,
+		struct sway_view *view, struct wlr_scene_tree *parent) {
 	struct wlr_xdg_surface *xdg_surface = wlr_popup->base;
 
-	struct sway_xdg_popup *popup =
-		calloc(1, sizeof(struct sway_xdg_popup));
-	if (popup == NULL) {
+	struct sway_xdg_popup *popup = calloc(1, sizeof(struct sway_xdg_popup));
```

Comment:
`sizeof(*popup)` (though I realize you didn't originally write this code, and it's only getting reformatted here)